drinks_touch/screens/screen.py

```diff
@@ -36,11 +36,17 @@ def render(self, dt):
         else:
             debug_surface = None
         for o in self.objects:
+            o.tick(dt)
             if not o.visible:
                 continue
-            obj_surface = o.render(dt)
-            if obj_surface is not None:
-                surface.blit(obj_surface, o.screen_pos)
+            if o.last_hash != o.calculate_hash():
+                o.dirty = True
+                o.last_hash = o.calculate_hash()
```

Comment:
reuse previous hash value

Reply:
Thanks, using now obj.render_cached(), to not repeat the mechanism in two places.